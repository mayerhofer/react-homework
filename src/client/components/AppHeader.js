import React from 'react';
import { Navbar } from 'react-bootstrap';

import * as pageTexts from '../../constants/pageTexts';

// const NavBody = (props) => (
//     <Nav>
//         <NavItem evenKey={1} href="#">
//         </NavItem>
//     </Nav>
// );

class AppHeader extends React.Component {
    navbarStyle = {
        background: 'black',
        backgroundColor: 'black',
        borderBottom: '0px',
        height: '100%',
        width: '100%',
    };

    linkStyle = {
        color: 'red',
    };

    navabarBrandStyle = { backgroundColor: 'black', marginLeft: '100px!important' }

    NavHeader = () => (
        <Navbar.Header>
            <Navbar.Brand style={this.navabarBrandStyle}>
                <h3><a style={this.linkStyle} href="#home">{pageTexts.CAPTION_HEADER_TITLE}</a></h3>
            </Navbar.Brand>
        </Navbar.Header>
    );

    render() {
        return (
            <Navbar style={this.navbarStyle}>
                <NavHeader />
                {/* <NavBody content={this.props.content} /> */}
            </Navbar>);
    }
}

export default AppHeader;
