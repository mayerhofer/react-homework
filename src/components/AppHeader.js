import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import { Nav, Navbar, NavItem, NavDropdown } from 'react-bootstrap';

const navbarStyle = {
    background: 'transparent',
    backgroundColor: 'transparent',
    borderColor: 'transparent',
}

const linkStyle = {
    color: 'red'
}

const NavHeader = (props) => (
    <Navbar.Header>
        <Navbar.Brand style={{marginLeft: '100px!important'}}><h3><a style={linkStyle} href="#home">{props.title}</a></h3></Navbar.Brand>
    </Navbar.Header>
);

// const NavBody = (props) => (
//     <Nav>
//         <NavItem evenKey={1} href="#">
//         </NavItem>
//     </Nav>
// );

class AppHeader extends React.Component {

    render() {
        return (
        <Navbar style={navbarStyle}>
            <NavHeader title={this.props.title} />
            {/* <NavBody content={this.props.content} /> */}
        </Navbar>);
    };
}

export default AppHeader;
