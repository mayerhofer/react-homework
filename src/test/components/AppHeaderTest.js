import React from 'react';
import { render } from 'enzyme';

import AppHeader from '../../client/components/AppHeader';

describe('AppHeader Component', () => {
    it('should render without throwing an error', () => {
        expect(render(<AppHeader />)).toMatchSnapshot();
    });
});
