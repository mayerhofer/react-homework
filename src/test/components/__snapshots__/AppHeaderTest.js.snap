// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppHeader Component should render without throwing an error 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "navbar navbar-default",
      "style": "background:transparent;background-color:transparent;border-color:transparent",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "container",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "navbar-header",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "class": "navbar-brand",
                },
                "children": Array [
                  Object {
                    "attribs": Object {
                      "href": "#home",
                      "style": "color:red",
                    },
                    "children": Array [],
                    "name": "a",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "href": undefined,
                      "style": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "href": undefined,
                      "style": undefined,
                    },
                  },
                ],
                "name": "h3",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "nav",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "style": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "style": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
